modules/manager/run.go

```diff
@@ -202,64 +204,67 @@ func Run(
 		UserName: cfg.ServiceAccountToImpersonate,
 	}
 
-	cacheOptions := cache.Options{}
+	var cacheOptions cache.Options
 	if cfg.CacheSyncPeriod > 0 {
 		setupLog.Info("cache sync period set", "period", cfg.CacheSyncPeriod)
 		cacheOptions.SyncPeriod = &cfg.CacheSyncPeriod
 	}
 
-	cacheOptions.ByObject = map[client.Object]cache.ByObject{}
-	if cfg.SecretLabelSelector != "" {
-		req, err := labels.NewRequirement(cfg.SecretLabelSelector, selection.Equals, []string{"true"})
-		if err != nil {
-			return fmt.Errorf("failed to make label requirement for secrets: %w", err)
-		}
-		cacheOptions.ByObject[&corev1.Secret{}] = cache.ByObject{
-			Label: labels.NewSelector().Add(*req),
+	// If there are no configured watch namespaces, then we're watching ALL namespaces,
+	// and we don't have to bother individually caching any particular namespaces.
+	// This is the default behavior of the controller-runtime manager.
+	// If there are configured watch namespaces, then we're watching only those namespaces.
+	if len(cfg.WatchNamespaces) > 0 {
+		watchNamespaces := strings.Split(cfg.WatchNamespaces, ",")
+		setupLog.Info("Manager set up with multiple namespaces", "namespaces", watchNamespaces)
+		watched := make(map[string]cache.Config)
+		for _, ns := range watchNamespaces {
+			ns = strings.TrimSpace(ns)
+			watched[ns] = cache.Config{}
 		}
+		cacheOptions.DefaultNamespaces = watched
 	}
 
-	if cfg.ConfigMapLabelSelector != "" {
-		req, err := labels.NewRequirement(cfg.ConfigMapLabelSelector, selection.Equals, []string{"true"})
-		if err != nil {
-			return fmt.Errorf("failed to make label requirement for config maps: %w", err)
+	if cfg.ConfigMapLabelSelector != "" || cfg.SecretLabelSelector != "" {
+		cacheOptions.ByObject = map[client.Object]cache.ByObject{}
+		if err := setByObjectFor[corev1.Secret](cfg.SecretLabelSelector, cacheOptions.ByObject); err != nil {
+			return fmt.Errorf("failed to set byObject for secrets: %w", err)
 		}
-		cacheOptions.ByObject[&corev1.ConfigMap{}] = cache.ByObject{
-			Label: labels.NewSelector().Add(*req),
+		if err := setByObjectFor[corev1.ConfigMap](cfg.ConfigMapLabelSelector, cacheOptions.ByObject); err != nil {
+			return fmt.Errorf("failed to set byObject for config maps: %w", err)
```

Comment:
```suggestion
			return fmt.Errorf("failed to set byObject for ConfigMaps: %w", err)
```

I checked K8s docs, and they're writing it like that, [see](https://kubernetes.io/docs/concepts/configuration/configmap/#configmap-object)